src/Core/Grid/Definition/Factory/CartRuleGridDefinitionFactory.php

```diff
@@ -0,0 +1,262 @@
+                    ->add((new SubmitRowAction('delete'))
+                    ->setName($this->trans('Delete', [], 'Admin.Actions'))
+                    ->setIcon('delete')
+                    ->setOptions([
+                        'method' => 'POST',
+                        'route' => 'admin_cart_rules_delete',
+                        'route_param_name' => 'cartRuleId',
+                        'route_param_field' => 'id_cart_rule',
+                        'confirm_message' => $this->trans(
+                            'Delete selected item?',
+                            [],
+                            'Admin.Notifications.Warning'
+                        ),
+                    ])
+                    ),
+            ])
+            );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getFilters(): FilterCollection
+    {
+        return (new FilterCollection())
+            ->add((new Filter('id_cart_rule', TextType::class))
+            ->setTypeOptions([
+                'required' => false,
+                'attr' => [
+                    'placeholder' => $this->trans('ID', [], 'Admin.Global'),
+                ],
+            ])
+            ->setAssociatedColumn('id_cart_rule')
+            )
+            ->add((new Filter('name', TextType::class))
+            ->setTypeOptions([
+                'required' => false,
+                'attr' => [
+                    'placeholder' => $this->trans('Name', [], 'Admin.Global'),
+                ],
+            ])
+            ->setAssociatedColumn('name')
+            )
+            ->add((new Filter('priority', TextType::class))
+            ->setTypeOptions([
+                'required' => false,
+                'attr' => [
+                    'placeholder' => $this->trans('Priority', [], 'Admin.Global'),
+                ],
+            ])
+            ->setAssociatedColumn('priority')
+            )
+            ->add((new Filter('code', TextType::class))
+            ->setTypeOptions([
+                'required' => false,
+                'attr' => [
+                    'placeholder' => $this->trans('Code', [], 'Admin.Global'),
+                ],
+            ])
+            ->setAssociatedColumn('code')
+            )
+            ->add((new Filter('quantity', TextType::class))
+            ->setTypeOptions([
+                'required' => false,
+                'attr' => [
+                    'placeholder' => $this->trans('Quantity', [], 'Admin.Catalog.Feature'),
+                ],
+            ])
+            ->setAssociatedColumn('quantity')
+            )
+            ->add((new Filter('date_to', DateRangeType::class))
+            ->setTypeOptions([
+                'required' => false,
+                'attr' => [
+                    'placeholder' => $this->trans('Expiration date', [], 'Admin.Catalog.Feature'),
+                ],
+                'date_format' => 'YYYY-MM-DD HH:mm:ss',
```

Comment:
this format differs from above one.

Reply:
I see that for other filters(in orders page this is the format used)

Reply:
You should refer to cart rules legacy page, not the orders behavior. 
Actually maybe @MatShir would know which date format should be used for cart rules page filter and form actions?

Reply:
@JevgenijVisockij Julius is right. Some pages use a format, other pages use another format. Sometimes for good reasons, and we must keep them (so we check legacy behavior and keep it) and sometimes for bad reasons (or no logical reasons). When we find bad or illogical reasons we challenge them with the help of @MatShir and we check if we cant find a better way.

Sometimes however we cannot change the setting even though it's bad or illogical because it creates important breaking changes that modules rely on.

So ... no "one rule fits all" here. Each item must be discussed and explored.

Reply:
In filter for legacy cart rule 
it's DD/MM/YYYY

@MatShir  Should I stick to that or should I change 'YYYY-MM-DD HH:mm:ss'?

Reply:
Let's stick to legacy for now 👍